explorer/scan/local_scanner.go

```diff
@@ -184,9 +190,34 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 		// we grab the asset from the connection, because it contains all the
 		// detected metadata (and IDs)
 		// assets = append(assets, runtime.Provider.Connection.Asset)
-		runtimes = append(runtimes, runtime)
 	}
 
+	// for each asset candidate, we initialize a new runtime and connect to it.
+	for _, asset := range assetCandidates {
```

Comment:
what if we keep a map of runtimes as we go through these, so that we can reuse an already created runtime if its been created?